code/modules/mob/dead/observer/observer.dm

```diff
@@ -95,7 +95,8 @@
 					name = capitalize(pick(first_names_female)) + " " + capitalize(pick(last_names))
 
 		mind = body.mind	//we don't transfer the mind but we keep a reference to it.
-
+	if(client && client.color)
```

Comment:
Login is called when being placed out of a new mob, being placed into an observer is such a case.